packages/nimble/R/externalCalls.R

```diff
@@ -0,0 +1,87 @@
+
+nimbleExternalCall <- function(fun, returnType = void(), Cfun, headerFile, oFile) {
```

Comment:
1. Passing in a `fun` seems like a hack. Why not simply pass in `name`, `argTypes`, and `returnType`?

2. Languages like Lua and Python provide a nice ffi libraries ([ffi](http://luajit.org/ext_ffi.html) and [cffi](https://cffi.readthedocs.io/en/latest/overview.html#simple-example-abi-level-in-line), resp), that allow users to specify the *text* of a header file, rather than a filename or even type signature. In our case, we should be able to *create* the proper header from `name`, `argTypes` and `returnType` if a header file is not provided.

3. I assume the `oFile` argument is the name of a shared library to link in? I believe this should a list `dllFiles` of multiple libraries. We may also want a `cppFiles` in case the user wants us to compile their code, so that they can leave this empty if they just want to call some internal R function or a C function that's part of another R package's internal implementation (in which case the package would already have been compiled and linked in), right?